recipes/runtimeqml/all/conanfile.py

```diff
@@ -0,0 +1,101 @@
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeToolchain, CMakeDeps, cmake_layout
+from conan.tools.microsoft import is_msvc, check_min_vs
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.files import get, copy
+from conan.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.50.0"
+
+class RuntimeQml(ConanFile):
+    name = "runtimeqml"
+    homepage = "https://github.com/GIPdA/runtimeqml"
+    description = "Enables hot-reloading qml files"
+    topics = ("qt", "hot-reload", "qml", "gui")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "BSD-3-Clause"
+    settings = "os", "arch", "compiler", "build_type"
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True
+    }
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "7",
+            "clang": "7",
+            "apple-clang": "10",
+        }
+
+    def export_sources(self):
+        copy(self, "CMakeLists.txt", self.recipe_folder, self.export_sources_folder)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+        self.options["qt"].qtdeclarative = True
```

Comment:
@uilianries Should I be doing this? as this seems to end up not letting the package "build".

Reply:
It occurred because `validate()` is executed after `requirements()`: https://docs.conan.io/en/latest/reference/commands/creator/create.html#methods-execution-order

You will need to remove it from `configure()`